hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -1902,9 +1901,9 @@ public NClob createNClob(Reader reader, long length) {
 		}
 	}
 
-	private static class SharedSessionBuilderImpl<T extends SharedSessionBuilder>
-			extends SessionFactoryImpl.SessionBuilderImpl<T>
-			implements SharedSessionBuilder<T>, SharedSessionCreationOptions {
+	private static class SharedSessionBuilderImpl
+			extends SessionFactoryImpl.SessionBuilderImpl<SharedSessionBuilderImpl>
+			implements SharedSessionBuilder<SharedSessionBuilderImpl>, SharedSessionCreationOptions {
 		private final SessionImpl session;
```

Comment:
The above class is the leaf classes in the hierarchy (it is private class in this class) and there is no purpose to provide a generic type here (which is purely meant for subclassing). We are supposed to enjoy the generics at last by replacing `T` with the terminal class name per se, as seen in the below code changes. Time to harvest!